crates/cdk/src/nuts/nut02.rs

```diff
@@ -112,6 +112,20 @@ impl Id {
             id: bytes[1..].try_into()?,
         })
     }
+
+    /// [`Id`] to u64
+    pub fn to_u64(&self) -> u64 {
+        let bytes = self.to_bytes();
+        let mut array = [0u8; 8];
+        array[..bytes.len()].copy_from_slice(&bytes);
+        u64::from_be_bytes(array)
+    }
```

Comment:
I don't think this matches the nut here https://github.com/cashubtc/nuts/blob/main/13.md#keyset-id?

Can we just reuse https://github.com/cashubtc/cdk/blob/cc5b2673674f3517d267490008f605dec3b48628/crates/cdk/src/nuts/nut02.rs#L117-L126

Reply:
Ah, this is a much better idea! Sorry for the garbage PR. I can take care of that test tho.

Reply:
Actually, I take it back. The existing TryFrom implementation returns a u64 with half of the least significant bytes truncated. Shouldn't it return a u32 instead of a u64?

I need functions to convert the 8 byte array to a u64 and back without data loss.

Reply:
I'm not sure what the existing TryFrom is used for, or if it is used. I think the simplest fix would be to update it to return a u32 and then write two new impls. Is this a good plan?
```
impl TryFrom<u64> for Id
impl TryFrom<Id> for u64
```

Reply:
I opened a PR in the nuts repo to clarify the language. https://github.com/cashubtc/nuts/pull/189